utilities/src/parallel.rs

```diff
@@ -169,12 +169,68 @@ macro_rules! cfg_chunks_mut {
 #[macro_export]
 macro_rules! cfg_reduce {
     ($e: expr, $default: expr, $op: expr) => {{
-        #[cfg(feature = "parallel")]
+        #[cfg(not(feature = "serial"))]
         let result = $e.reduce($default, $op);
 
-        #[cfg(not(feature = "parallel"))]
+        #[cfg(feature = "serial")]
         let result = $e.fold($default(), $op);
 
         result
     }};
 }
+
+/// Applies `reduce_with` or `reduce` depending on the `serial` feature.
+#[macro_export]
+macro_rules! cfg_reduce_with {
+    ($e: expr, $op: expr) => {{
+        #[cfg(not(feature = "serial"))]
+        let result = $e.reduce_with($op);
+
+        #[cfg(feature = "serial")]
+        let result = $e.reduce($op);
+
+        result
+    }};
+}
+
+/// Turns a collection into an iterator.
+#[macro_export]
+macro_rules! cfg_values {
+    ($e: expr) => {{
+        #[cfg(not(feature = "serial"))]
+        let result = $e.par_values();
+
+        #[cfg(feature = "serial")]
+        let result = $e.values();
+
+        result
+    }};
+}
+
+/// Finds the first element that satisfies the predicate function
+#[macro_export]
+macro_rules! cfg_find {
+    ($self:ident, $object:expr, $func:ident) => {{
+        #[cfg(not(feature = "serial"))]
+        let result = $self.par_values().find_any(|tx| tx.$func($object));
+
+        #[cfg(feature = "serial")]
+        let result = $self.values().find(|tx| tx.$func($object));
+
+        result
+    }};
+}
+
+/// Applies a function and returns the first value that is not None
+#[macro_export]
+macro_rules! cfg_find_map {
+    ($self:ident, $object:expr, $func:ident) => {{
+        #[cfg(not(feature = "serial"))]
+        let result = $self.par_values().filter_map(|tx| tx.$func($object)).find_any(|_| true);
```

Comment:
```suggestion
        let result = $self.par_values().filter_map(|tx| tx.$func($object)).take_any(1);
```
I think this would make it a bit more clear

Reply:
The type signature isn't the same and trying to make that work complicates the iterator a bit more